homeassistant/components/keenetic_ndms2/config_flow.py

```diff
@@ -48,9 +50,8 @@ async def async_step_user(self, user_input=None):
         """Handle a flow initialized by the user."""
         errors = {}
         if user_input is not None:
-            for entry in self.hass.config_entries.async_entries(DOMAIN):
-                if entry.data[CONF_HOST] == user_input[CONF_HOST]:
-                    return self.async_abort(reason="already_configured")
+            if self._already_configured(user_input[CONF_HOST]):
+                return self.async_abort(reason="already_configured")
```

Comment:
```suggestion
            await self.async_set_unique_id(user_input[CONF_HOST])
            self._abort_if_unique_id_configured() 
```

Reply:
Hold of on implementing this change. I think it might be a bad idea actually. Setting the unique id to something read over the telnet connection would be much better.